server/routerlicious/packages/services-utils/src/executeOnInterval.ts

```diff
@@ -0,0 +1,28 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Lumberjack } from "@fluidframework/server-services-telemetry";
+
+export async function executeOnInterval<T>(
+    api: () => Promise<T>,
+    intervalInMs: number,
+    callName: string): Promise<void> {
+    await api()
+        .then((res) => {
+            Lumberjack.info(`Success executing ${callName}`, undefined);
+        })
+        .catch((error) => {
+            Lumberjack.error(`Error running ${callName}`, undefined, error);
+        })
+        .finally(() => {
+            if (intervalInMs) {
+                const timeoutP = async () => new Promise((resolve) => {
+                    setTimeout(resolve, intervalInMs);
+                });
+                // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                timeoutP().then(async () => executeOnInterval(api, intervalInMs, callName));
```

Comment:
I'm not sure if this would be correct, but would it make sense to add `await` in front of `timeoutP().then(...)`? Is there no alternative other than suppressing the floating promise?